pipeline/splicing/calcVarPriors.py

```diff
@@ -1020,70 +1190,111 @@ def compareRefAltExonLengths(refLength, altLength):
     else:
         return False
 
-def isSplicingWindowInFrame(variant, exonicPortionSize, deNovoDonorInRefAcc=False):
+def isSplicingWindowInFrame(variant, exonicPortionSize, intronicPortionSize, deNovoDonorInRefAcc=False, donor=True):
     '''
     Given a variant, determines ref and alt exon length and compares them
     exonicPortionSize refers to length in bp that is considered to be in exonic portion of splice site
+    intronicPortionSize referes to length in bp that is considered to be in intronic portion of splice site
     deNovoDonorInRefAcc=True if looking for deNovoDonor in ref acceptor site, False otherwise
+    Donor argument determines if function is used for de novo donor (donor=True) or de novo acceptor (donor=False)
     If ref and alt exon are in the same reading frame, returns True
     '''
-    refLength = getRefExonLength(variant)
-    altLength = getAltExonLength(variant, exonicPortionSize, deNovoDonorInRefAcc=deNovoDonorInRefAcc)
+    refLength = getRefExonLength(variant, donor=donor)
+    altLength = getAltExonLength(variant, exonicPortionSize, intronicPortionSize, deNovoDonorInRefAcc=deNovoDonorInRefAcc, donor=donor)
     return compareRefAltExonLengths(refLength, altLength)
 
-def compareDeNovoWildTypeSplicePos(variant, exonicPortionSize, deNovoDonorInRefAcc=False):
+def compareDeNovoWildTypeSplicePos(variant, exonicPortionSize, intronicPortionSize, deNovoDonorInRefAcc=False, donor=True):
     '''
     Given a variant, compares de novo splicing position with wild-type splicing position
     exonicPortionSize refers to length in bp that is considered to be in exonic portion of splice site
+    intronicPortionSize referes to length in bp that is considered to be in intronic portion of splice site
     deNovoDonorInRefAcc argument=True if looking for de novo donor in reference splice acceptor region, False otherwise
+    Donor argument determines if function is used for de novo donor (donor=True) or de novo acceptor (donor=False)
     If distance between de novo and wild-type donors is divisible by 3, returns True
```

Comment:
The donor argument needs to be specified because if it is a de novo acceptor the new splice position has to be compared to the exon start (reference splice acceptor position) and for a de novo donor the new splice position has to be compared to the exon end. The function works for both and it is a way to check to see if splicing would cause a frameshift. The purpose of the function is to determine if the distance between reference and de novo splice positions is divisible by 3 but I didn't think `isDeNovoWildTypeSplicePosDistanceDivisibleByThree` was a good name. Do you have any suggestions for a better name?

Reply:
isDeNovoWildTypeSplicePosDistanceDivisibleByThree is actually fine even if it looks bonkers since it accurately describes what the function does. I still don't understand why we want to know if the distance between de novo and wild-type donors is divisible by 3 -- maybe an explanation could be included in the comment?

Reply:
I added an explanation in the comment, let me know if you want it to be more detailed